### PR TITLE
PhysicsBall Ball Friction Model

### LINKED ISSUE #1609: Initialize parameters of LinearBallModel using physics inputs
### Description of the task
The parameters of the LinearBallModel are currently 0 as a default state with no friction. The parameters should be calculated from physics inputs, such as coefficients of friction and moment of inertia. These should be injected in an easy-to-change framework. Physics explanation: https://physics.stackexchange.com/questions/114191/help-with-a-solid-sphere-sliding-and-then-rolling

### Acceptance criteria

- [ ] framework for injecting parameters
- [ ] calculating parameters from physics input
- [ ] system for changing/calibrating physics inputs

### LINKED ISSUE #1759: Support Ball Model in Simulation Physics
### Description of the task
Current state of ball friction in simulation: for the standalone simulator, we add linear damping to the ball to make it "look nice", but for simulated tests, we don't have any linear damping at all

In order to properly add friction to simulation, we should make the simulation match the ball model. This will allow our AI and simulation to work together perfectly and we can develop gameplay without worrying about a mismatch in ball model. 

Steps: 
1. make a BallModelConfig to dependency inject into the BallModel 
2. make the simulator own a BallModel so that it can use the BallModel to simulate the BallModel
3. Implement the physics to match the BallModel

### Acceptance criteria

- [ ] make a BallModelConfig to dependency inject into the BallModel 
- [ ] make the simulator own a BallModel so that it can use the BallModel to simulate the BallModel
- [ ] Implement the physics to match the BallModel

### LINKED ISSUE #1771: Make ball restitution and linear damping dynamic parameters
### Description of the task
Ball restitution and linear damping from the `Simulator` constructor
```
    explicit Simulator(const Field& field, double ball_restitution,
                       double ball_linear_damping,
                       const Duration& physics_time_step =
                           Duration::fromSeconds(DEFAULT_PHYSICS_TIME_STEP_SECONDS));
```
should be dynamic parameters so we can properly tune them to "look right". These values affect how the ball bounces off the walls and robots and how it slows down as it travels

### Acceptance criteria

- [ ] Put these values in a Simulator Config
- [ ] Tune the values so it looks about right

### PR DESCRIPTION
**Pull request by jonathanlew**
### Description
* Added a sliding/rolling friction model to PhysicsBall
* Friction acceleration values set up for StandaloneSimulator
* Added SimulatorConfig to set friction acceleration values
* Added SimulatorConfig widget to the StandaloneSimulator

### Testing Done
added unit tests to applyBallFrictionModel, tested StandaloneSimulator GUI elements manually

### Resolved Issues
resolves #1771
resolves #1759 
resolves #1609 

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [ ] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [ ] **Remove all commented out code**
- [ ] **Remove extra print statements**: for example, those just used for testing
- [ ] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [ ] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

File: src/software/simulation/physics/physics_ball.cpp
Comment (kelvinkoon) [inline src/software/simulation/physics/physics_ball.cpp]: Commented out code. Please remove.

File: src/software/simulation/physics/physics_ball.cpp
Comment (kelvinkoon) [inline src/software/simulation/physics/physics_ball.cpp:213]: Where does `5.0 / 7.0` come from?

File: src/software/simulation/physics/physics_ball.cpp
Comment (kelvinkoon) [inline src/software/simulation/physics/physics_ball.cpp:255]: Quick check, do we ever have to consider dividing by 0 here?

File: src/software/simulation/physics/physics_ball.h
Comment (kelvinkoon) [inline src/software/simulation/physics/physics_ball.h]: "is needs" -> "is needed"

**Review by kelvinkoon — CHANGES_REQUESTED** [review]
:soccer:

File: src/software/world/ball.cpp
Comment (itsarune) [inline src/software/world/ball.cpp]: I think you meant "than" here

File: src/software/parameter/config/simulator.yaml
Comment (itsarune) [inline src/software/parameter/config/simulator.yaml:5]: adding to @kelvinkoon 's comment, wouldn't sliding friction being defaulted to 0 be problematic?

File: src/software/simulation/standalone_simulator_main.cpp
Comment (akhilveeraghanta) [inline src/software/simulation/standalone_simulator_main.cpp:39]: we can probably just put these in the yaml as default values?

File: src/software/simulation/physics/physics_ball.cpp
Comment (akhilveeraghanta) [inline src/software/simulation/physics/physics_ball.cpp]: I think its worth summarizing how you achieved this from reading that paper

**Review by akhilveeraghanta — COMMENTED** [review]
overall lgtm! its really nice to see the ball actually slow down, I played around with really horrible edge cases and it seems like the friction model is pretty robust

nice work as usual

File: src/software/simulation/standalone_simulator_main.cpp
Comment (jonathanlew) [reply src/software/simulation/standalone_simulator_main.cpp:39]: hm, the physicsball, physicsworld, simulator tests all depend on friction being 0

File: src/software/world/ball.cpp
Comment (jonathanlew) [reply src/software/world/ball.cpp]: You're finding a lot of spelling mistakes in our existing code. It'd be great if you could put  up a PR to fix a bunch of them at once. (I'll fix this one though)

File: src/software/simulation/physics/physics_ball.cpp
Comment (jonathanlew) [reply src/software/simulation/physics/physics_ball.cpp]: added a comment

File: src/software/parameter/config/simulator.yaml
Comment (jonathanlew) [reply src/software/parameter/config/simulator.yaml:5]: added a comment

File: src/software/simulation/physics/physics_ball.cpp
Comment (jonathanlew) [reply src/software/simulation/physics/physics_ball.cpp:213]: added comment

File: src/software/simulation/physics/physics_ball.cpp
Comment (jonathanlew) [reply src/software/simulation/physics/physics_ball.cpp]: removed

**Review by kelvinkoon — APPROVED** [review]
LGTM pending existing comments.

File: src/software/simulation/standalone_simulator_main.cpp
Comment (akhilveeraghanta) [reply src/software/simulation/standalone_simulator_main.cpp:39]: ok fair enough, id probably reset it in the test fixture rather than here, but its more a personal preference thing so won't block on this

**Review by akhilveeraghanta — APPROVED** [review]

**Pull request merged**
